network/src/network.rs

```diff
@@ -468,31 +469,43 @@ impl NetworkState {
     /// this method is intent to check observed addr by dial to self
     pub(crate) fn try_dial_observed_addrs(&self, p2p_control: &ServiceControl) {
         let mut pending_observed_addrs = self.pending_observed_addrs.write();
-        let public_addrs = { self.public_addrs.read().clone() };
-        for addr in pending_observed_addrs
-            .drain()
-            .chain(public_addrs.into_iter())
-        {
+        if pending_observed_addrs.is_empty() {
+            let addrs = self.public_addrs.write();
```

Comment:
Why dial `self.public_addrs` in a `fn try_dial_observed_addrs()` ?

Reply:
the observed address is not always valid. For example, the address obtained through the reuse port extension may become invalid as the connection is disconnected. This part of the address information needs to be cleaned up.